metric_name_builder.go

```diff
@@ -116,7 +116,7 @@ func BuildCompliantMetricName(metric pmetric.Metric, namespace string, addMetric
 
 var (
 	// Regexp for metric name characters that should be replaced with _.
-	invalidMetricCharRE   = regexp.MustCompile(`[^a-zA-Z0-9:_]`)
+	invalidMetricCharRE   = regexp.MustCompile(`[^a-zA-Z0-9:]`)
```

Comment:
Can you add the following explanation after the comment on line 99? The behaviour of "compacting" underscores is already documented like this for the `addMetricSuffixes` case (i.e. in `normalizeName`).

```golang
	// Note that this has the side effect of replacing multiple consecutive underscores with a single underscore.
	// This is part of the OTel to Prometheus specification: https://github.com/open-telemetry/opentelemetry-specification/blob/v1.38.0/specification/compatibility/prometheus_and_openmetrics.md#otlp-metric-points-to-prometheus.
```

Reply:
done!